lib/linguist/heuristics.yml

```
@@ -360,6 +360,14 @@ disambiguations:
   rules:
   - language: Unity3D Asset
     pattern: 'tag:unity3d.com'
+- extensions: ['.mc']
+  rules:
+  - language: Win32 Message File
+    pattern: '(?i)MessageId=|^;'
+  - language: M4
+    pattern: '^dnl|^divert\(|^\w+\(`'
```

```suggestion
    pattern: '^dnl|^divert\((?:-?\d+)?\)|^\w+\(`[^\n]*?'[),]'
```

1. IIRC, M4's [`divert()` built-in](https://web.mit.edu/gnu/doc/html/m4_9.html#SEC45) only accepts a numeric argument.

2. Traditional implementations of M4 use [\`old-school typewriter quotes'](https://github.com/Alhadis/Atom-XTermQuotes) for delimiting macro arguments.